# Santander Coders - Ciência de Dados (Ada Tech)  
# Trabalho Final - Técnicas de Programação I  

### Aluno: Gabriel Azevedo Anunciação

## Importações das bibliotecas e carregamento dos dados

In [231]:
import pandas as pd
import numpy as np

link_1 = "https://github.com/gabriel7268/Santander_Coders2023.2-Data_Science/raw/main/Datasets/olist_customers_dataset.csv"
link_2 = "https://github.com/gabriel7268/Santander_Coders2023.2-Data_Science/raw/main/Datasets/olist_order_items_dataset.csv"
link_3 = "https://github.com/gabriel7268/Santander_Coders2023.2-Data_Science/raw/main/Datasets/olist_order_payments_dataset.csv"

links = np.array([link_1,link_2,link_3])

df_customers = pd.read_csv(links[0])
df_items = pd.read_csv(links[1])
df_payments = pd.read_csv(links[2])

## Tabela de Clientes


In [232]:
display(df_customers.head())
print(f"Shape: {df_customers.shape}")
print(f"Tipo: {type(df_customers)}")
print(f"Tamanho em bytes: {df_customers.to_numpy().nbytes}")

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

Shape: (99441, 5)
Tipo: <class 'pandas.core.frame.DataFrame'>
Tamanho em bytes: 3977640


Esta tabela, do tipo **Pandas DataFrame**, contém **5 colunas**, as quais dão informações sobre o **ID do cliente, o ID único do cliente, o prefixo do código postal do cliente, a cidade e o estado do cliente**.  

A partir dela podemos identificar que há **99.441 clientes**.


Podemos alterar o nome das colunas, para facilitar a visualização:

In [233]:
print(df_customers.columns)

Index(['customer_id', 'customer_unique_id', 'customer_zip_code_prefix',
       'customer_city', 'customer_state'],
      dtype='object')


In [234]:
df_customers.columns = ['ID Cliente', 'ID Único Cliente', 'Prefixo Código Postal',
                        'Cidade', 'Estado']
display(df_customers.head())

ID Cliente                  ID Único Cliente  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   Prefixo Código Postal                 Cidade Estado  
0                  14409                 franca     SP  
1                   9790  sao bernardo do campo     SP  
2                   1151              sao paulo     SP  
3                   8775        mogi das cruzes     SP  
4                  13056               campinas     SP

In [235]:
display(df_customers.isnull().sum())

ID Cliente               0
ID Único Cliente         0
Prefixo Código Postal    0
Cidade                   0
Estado                   0
dtype: int64

A partir da célula acima é possível perceber que **não há dados nulos** em nenhuma linha de nenhuma coluna do DataFrame de clientes.

Vamos, agora, analisar a diferença entre as colunas "ID Cliente" e "ID Único Cliente".

In [236]:
IDUnicoCliente_qnt = df_customers['ID Único Cliente'].unique().size
IDCliente_qnt = df_customers['ID Cliente'].unique().size
print(f"O número de itens únicos da tabela 'ID Cliente' é: {IDCliente_qnt}")
print(f"O número de itens únicos da tabela 'ID Único Cliente' é: {IDUnicoCliente_qnt}")

O número de itens únicos da tabela 'ID Cliente' é: 99441
O número de itens únicos da tabela 'ID Único Cliente' é: 96096


Pode-se perceber que a coluna "ID Cliente" tem 99.441 valores únicos, exatamente igual ao número de clientes, portanto esse ID é realmente único, enquanto que a coluna "ID Único Cliente" tem 96096, pela análise anterior verificamos que não há dados nulos, portanto há, nesta coluna, valores duplicados, portanto eliminares essa coluna da tabela, e trabalharemos apenas com a coluna "ID Cliente" e ela será o indíce da tabela.

In [237]:
df_customers.drop('ID Único Cliente', axis = 1, inplace = True)
df_customers.set_index('ID Cliente', inplace=True)

In [238]:
display(df_customers.head())

Prefixo Código Postal  \
ID Cliente                                                
06b8999e2fba1a1fbc88172c00ba8bc7                  14409   
18955e83d337fd6b2def6b18a428ac77                   9790   
4e7b3e00288586ebd08712fdd0374a03                   1151   
b2b6027bc5c5109e529d4dc6358b12c3                   8775   
4f2d8ab171c80ec8364f7c12e35b23ad                  13056   

                                                 Cidade Estado  
ID Cliente                                                      
06b8999e2fba1a1fbc88172c00ba8bc7                 franca     SP  
18955e83d337fd6b2def6b18a428ac77  sao bernardo do campo     SP  
4e7b3e00288586ebd08712fdd0374a03              sao paulo     SP  
b2b6027bc5c5109e529d4dc6358b12c3        mogi das cruzes     SP  
4f2d8ab171c80ec8364f7c12e35b23ad               campinas     SP

Podemos buscar um cliente específico e saber de qual cidade ele é:

In [239]:
display(df_customers.at['4f2d8ab171c80ec8364f7c12e35b23ad','Cidade'])

'campinas'

Podemos ordenar a tabela em ordem alfabética das cidades:

In [240]:
display(df_customers.sort_values('Cidade'))

Prefixo Código Postal               Cidade  \
ID Cliente                                                                     
a23e3f9a2b656b23b7e52075964b42cd                  38540  abadia dos dourados   
9e01f714a2b3b8962c222cf2b74c20dc                  38540  abadia dos dourados   
f11eb8f0b8b87510a93e3e1aa10b0ade                  38540  abadia dos dourados   
576d71ddb21b21763cfedce73b902180                  72940            abadiania   
5e9e1ae42e02df93e9a591e86fd531a3                  35620               abaete   
...                                                 ...                  ...   
1052e334b46dd1e9876946d6372a363e                  65365              ze doca   
d8b3ef6f73de33ae716e94d2478cc080                  65365              ze doca   
fc87e5c06780700952aa42998a22968f                  65365              ze doca   
0a466d490a3c2fcc1e7177f4809ec7dd                  89633               zortea   
e046f992a304cc92e1b00d7fccdf1c42                  89633               zortea   

                                 Estado  
ID Cliente                               
a23e3f9a2b656b23b7e52075964b42cd     MG  
9e01f714a2b3b8962c222cf2b74c20dc     MG  
f11eb8f0b8b87510a93e3e1aa10b0ade     MG  
576d71ddb21b21763cfedce73b902180     GO  
5e9e1ae42e02df93e9a591e86fd531a3     MG  
...                                 ...  
1052e334b46dd1e9876946d6372a363e     MA  
d8b3ef6f73de33ae716e94d2478cc080     MA  
fc87e5c06780700952aa42998a22968f     MA  
0a466d490a3c2fcc1e7177f4809ec7dd     SC  
e046f992a304cc92e1b00d7fccdf1c42     SC  

[99441 rows x 3 columns]

Podemos verificar a quantidade de clientes por cidade, em ordem decrescente para verificar as cidades com maior número de clientes:

In [241]:
display(df_customers.drop('Prefixo Código Postal', axis = 1, inplace = False).groupby('Cidade').value_counts().sort_values(ascending=False))

Cidade            Estado
sao paulo         SP        15540
rio de janeiro    RJ         6882
belo horizonte    MG         2773
brasilia          DF         2131
curitiba          PR         1521
                            ...  
ewbank da camara  MG            1
marechal deodoro  AL            1
fagundes          PB            1
farol             PR            1
nova laranjeiras  PR            1
Length: 4310, dtype: int64

Podemos verificar a quantidade de clientes por estado em ordem decrescente:

In [242]:
display(df_customers.drop(['Prefixo Código Postal','Cidade'], axis = 1, inplace = False).groupby('Estado').value_counts().sort_values(ascending=False))

Estado
SP    41746
RJ    12852
MG    11635
RS     5466
PR     5045
SC     3637
BA     3380
DF     2140
ES     2033
GO     2020
PE     1652
CE     1336
PA      975
MT      907
MA      747
MS      715
PB      536
PI      495
RN      485
AL      413
SE      350
TO      280
RO      253
AM      148
AC       81
AP       68
RR       46
dtype: int64

Podemos ver, quais são os estados que tem mais de 1.000 clientes, e quais não.

In [243]:
display(df_customers.drop(['Prefixo Código Postal','Cidade'], axis = 1, inplace = False).groupby('Estado').value_counts() > 1000)

Estado
AC    False
AL    False
AM    False
AP    False
BA     True
CE     True
DF     True
ES     True
GO     True
MA    False
MG     True
MS    False
MT    False
PA    False
PB    False
PE     True
PI    False
PR     True
RJ     True
RN    False
RO    False
RR    False
RS     True
SC     True
SE    False
SP     True
TO    False
dtype: bool

Podemos, também, verificar quais são as cidades, do estado RJ, em que há clientes:

In [244]:
display(df_customers[df_customers['Estado'] == 'RJ']['Cidade'].unique())

array(['rio de janeiro', 'resende', 'parati', 'sao goncalo',
       'nova iguacu', 'campos dos goytacazes', 'vassouras', 'niteroi',
       'paraiba do sul', 'cabo frio', 'macae', 'itatiaia',
       'duque de caxias', 'varre-sai', 'mangaratiba', 'petropolis',
       'belford roxo', 'volta redonda', 'barra mansa', 'itaguai', 'pirai',
       'sao joao de meriti', 'aperibe', 'saquarema', 'nova friburgo',
       'italva', 'tres rios', 'sao pedro da aldeia', 'rio das ostras',
       'passa tres', 'itaborai', 'teresopolis', 'ipiabas', 'valenca',
       'angra dos reis', 'sao jose de uba', 'guapimirim', 'mage', 'carmo',
       'armacao dos buzios', 'santo antonio de padua', 'queimados',
       'marica', 'quissama', 'araruama', 'comendador levy gasparian',
       'casimiro de abreu', 'trajano de moraes', 'rio bonito', 'cambuci',
       'vila muriqui', 'cordeiro', 'barra do pirai', 'sao joao da barra',
       'cachoeiras de macacu', 'japeri', 'seropedica', 'monnerat',
       'iguaba grande', 'mi

In [245]:
display(df_customers[df_customers['Estado'] == 'RJ']['Cidade'].unique().size)

149

É possível perceber que há, no Rio de Janeiro, 149 cidades em que há clientes.

## Tabela Itens

In [246]:
display(df_items)
print(f"Shape: {df_items.shape}")
print(f"Tipo: {type(df_items)}")
print(f"Tamanho em bytes: {df_items.to_numpy().nbytes}")

order_id  order_item_id  \
0       00010242fe8c5a6d1ba2dd792cb16214              1   
1       00018f77f2f0320c557190d7a144bdd3              1   
2       000229ec398224ef6ca0657da4fc703e              1   
3       00024acbcdf0a6daa1e931b038114c75              1   
4       00042b26cf59d7ce69dfabb4e55b4fd9              1   
...                                  ...            ...   
112645  fffc94f6ce00a00581880bf54a75a037              1   
112646  fffcd46ef2263f404302a634eb57f7eb              1   
112647  fffce4705a9662cd70adb13d4a31832d              1   
112648  fffe18544ffabc95dfada21779c9644f              1   
112649  fffe41c64501cc87c801fd61db3f6244              1   

                              product_id                         seller_id  \
0       4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1       e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2       c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3       7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4       ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   
...                                  ...                               ...   
112645  4aa6014eceb682077f9dc4bffebc05b0  b8bc237ba3788b23da09c0f1f3a3288c   
112646  32e07fd915822b0765e448c4dd74c828  f3c38ab652836d21de61fb8314b69182   
112647  72a30483855e2eafc67aee5dc2560482  c3cfdc648177fdbbbb35635a37472c53   
112648  9c422a519119dcad7575db5af1ba540e  2b3e4a2a3ea8e01938cabda2a3e5cc79   
112649  350688d9dc1e75ff97be326363655e01  f7ccf836d21b2fb1de37564105216cc1   

        shipping_limit_date   price  freight_value  
0       2017-09-19 09:45:35   58.90          13.29  
1       2017-05-03 11:05:13  239.90          19.93  
2       2018-01-18 14:48:30  199.00          17.87  
3       2018-08-15 10:10:18   12.99          12.79  
4       2017-02-13 13:57:51  199.90          18.14  
...                     ...     ...            ...  
112645  2018-05-02 04:11:01  299.99          43.41  
112646  2018-07-20 04:31:48  350.00          36.53  
112647  2017-10-30 17:14:25   99.90          16.95  
112648  2017-08-21 00:04:32   55.99           8.72  
112649  2018-06-12 17:10:13   43.00          12.79  

[112650 rows x 7 columns]

Shape: (112650, 7)
Tipo: <class 'pandas.core.frame.DataFrame'>
Tamanho em bytes: 6308400


Esta tabela, por sua vez, contém informações de ID das ordens, Quantidade de itens das ordens, ID dos produtos, ID dos vendedores, data limite de embarque, preço do produto, preço do frete.

In [247]:
print(df_items.columns)

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='object')


Renomeando os nomes das colunas, temos:

In [248]:
df_items.columns = ['ID Ordem', 'Qtd Itens', 'ID Produto', 'ID Vendedor',
       'Data Limite de Embarque', 'Preço', 'Frete']
display(df_items.head())

ID Ordem  Qtd Itens  \
0  00010242fe8c5a6d1ba2dd792cb16214          1   
1  00018f77f2f0320c557190d7a144bdd3          1   
2  000229ec398224ef6ca0657da4fc703e          1   
3  00024acbcdf0a6daa1e931b038114c75          1   
4  00042b26cf59d7ce69dfabb4e55b4fd9          1   

                         ID Produto                       ID Vendedor  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

  Data Limite de Embarque   Preço  Frete  
0     2017-09-19 09:45:35   58.90  13.29  
1     2017-05-03 11:05:13  239.90  19.93  
2     2018-01-18 14:48:30  199.00  17.87  
3     2018-08-15 10:10:18   12.99  12.79  
4     2017-02-13 13:57:51  199.90  18.14

Podemos verificar a quantidade de produtos vendidos por ordem, apresentando a tabela ordenada de forma descendente pela quantidade de itens por ordem.

In [249]:
display(df_items.sort_values('Qtd Itens', ascending = False))

ID Ordem  Qtd Itens  \
57317   8272b63d03f5f79c56e9e4120aec44ef         21   
57316   8272b63d03f5f79c56e9e4120aec44ef         20   
11951   1b15974a0141d54e36626dca3fdc731a         20   
75122   ab14fdcfbe524636d65ee38360e22ce8         20   
75121   ab14fdcfbe524636d65ee38360e22ce8         19   
...                                  ...        ...   
40059   5b2f8cfd2fc117b59e6a0247d77f07d1          1   
40058   5b2f1ff03b77977af4258e7cfc2ec799          1   
40057   5b2e79428198e5ce482f9ec1bd1014c3          1   
40056   5b2e70f861de8953b06ea667adc2eb4e          1   
112649  fffe41c64501cc87c801fd61db3f6244          1   

                              ID Produto                       ID Vendedor  \
57317   79ce45dbc2ea29b22b5a261bbb7b7ee7  2709af9587499e95e803a6498a5a56e9   
57316   270516a3f41dc035aa87d220228f844c  2709af9587499e95e803a6498a5a56e9   
11951   ee3d532c8a438679776d222e997606b3  8e6d7754bc7e0f22c96d255ebda59eba   
75122   9571759451b1d780ee7c15012ea109d4  ce27a3cc3c8cc1ea79d11e561e9bebb6   
75121   9571759451b1d780ee7c15012ea109d4  ce27a3cc3c8cc1ea79d11e561e9bebb6   
...                                  ...                               ...   
40059   4a16cff2de159237df876ff7ba0fceca  9cf787a239c1aa29dbd76f153dc13f9a   
40058   b5adf7b72a7eb667aafd446581c35f3f  c60b801f2d52c7f7f91de00870882a75   
40057   2029248294c186017ecfc48cecd47bbd  d50d79cb34e38265a8649c383dcffd48   
40056   cb80d84cdedee288c467491ed98f5e1f  8ab42aa58097fd4668d60cc648225d5f   
112649  350688d9dc1e75ff97be326363655e01  f7ccf836d21b2fb1de37564105216cc1   

       Data Limite de Embarque   Preço  Frete  
57317      2017-07-21 18:25:23    7.80   6.57  
57316      2017-07-21 18:25:23    1.20   7.89  
11951      2018-03-01 02:50:48  100.00  10.12  
75122      2017-08-30 14:30:23   98.70  14.44  
75121      2017-08-30 14:30:23   98.70  14.44  
...                        ...     ...    ...  
40059      2018-06-13 17:53:44   79.00  12.99  
40058      2018-05-03 13:51:08  241.00  19.57  
40057      2017-07-17 04:15:27  126.99  48.33  
40056      2018-05-04 18:29:54  235.00  23.71  
112649     2018-06-12 17:10:13   43.00  12.79  

[112650 rows x 7 columns]

Tornando a coluna ID Ordem como índice do DataFrame, temos:

In [250]:
df_items.set_index('ID Ordem', inplace=True)
display(df_items)

Qtd Itens                        ID Produto  \
ID Ordem                                                                        
00010242fe8c5a6d1ba2dd792cb16214          1  4244733e06e7ecb4970a6e2683c13e61   
00018f77f2f0320c557190d7a144bdd3          1  e5f2d52b802189ee658865ca93d83a8f   
000229ec398224ef6ca0657da4fc703e          1  c777355d18b72b67abbeef9df44fd0fd   
00024acbcdf0a6daa1e931b038114c75          1  7634da152a4610f1595efa32f14722fc   
00042b26cf59d7ce69dfabb4e55b4fd9          1  ac6c3623068f30de03045865e4e10089   
...                                     ...                               ...   
fffc94f6ce00a00581880bf54a75a037          1  4aa6014eceb682077f9dc4bffebc05b0   
fffcd46ef2263f404302a634eb57f7eb          1  32e07fd915822b0765e448c4dd74c828   
fffce4705a9662cd70adb13d4a31832d          1  72a30483855e2eafc67aee5dc2560482   
fffe18544ffabc95dfada21779c9644f          1  9c422a519119dcad7575db5af1ba540e   
fffe41c64501cc87c801fd61db3f6244          1  350688d9dc1e75ff97be326363655e01   

                                                       ID Vendedor  \
ID Ordem                                                             
00010242fe8c5a6d1ba2dd792cb16214  48436dade18ac8b2bce089ec2a041202   
00018f77f2f0320c557190d7a144bdd3  dd7ddc04e1b6c2c614352b383efe2d36   
000229ec398224ef6ca0657da4fc703e  5b51032eddd242adc84c38acab88f23d   
00024acbcdf0a6daa1e931b038114c75  9d7a1d34a5052409006425275ba1c2b4   
00042b26cf59d7ce69dfabb4e55b4fd9  df560393f3a51e74553ab94004ba5c87   
...                                                            ...   
fffc94f6ce00a00581880bf54a75a037  b8bc237ba3788b23da09c0f1f3a3288c   
fffcd46ef2263f404302a634eb57f7eb  f3c38ab652836d21de61fb8314b69182   
fffce4705a9662cd70adb13d4a31832d  c3cfdc648177fdbbbb35635a37472c53   
fffe18544ffabc95dfada21779c9644f  2b3e4a2a3ea8e01938cabda2a3e5cc79   
fffe41c64501cc87c801fd61db3f6244  f7ccf836d21b2fb1de37564105216cc1   

                                 Data Limite de Embarque   Preço  Frete  
ID Ordem                                                                 
00010242fe8c5a6d1ba2dd792cb16214     2017-09-19 09:45:35   58.90  13.29  
00018f77f2f0320c557190d7a144bdd3     2017-05-03 11:05:13  239.90  19.93  
000229ec398224ef6ca0657da4fc703e     2018-01-18 14:48:30  199.00  17.87  
00024acbcdf0a6daa1e931b038114c75     2018-08-15 10:10:18   12.99  12.79  
00042b26cf59d7ce69dfabb4e55b4fd9     2017-02-13 13:57:51  199.90  18.14  
...                                                  ...     ...    ...  
fffc94f6ce00a00581880bf54a75a037     2018-05-02 04:11:01  299.99  43.41  
fffcd46ef2263f404302a634eb57f7eb     2018-07-20 04:31:48  350.00  36.53  
fffce4705a9662cd70adb13d4a31832d     2017-10-30 17:14:25   99.90  16.95  
fffe18544ffabc95dfada21779c9644f     2017-08-21 00:04:32   55.99   8.72  
fffe41c64501cc87c801fd61db3f6244     2018-06-12 17:10:13   43.00  12.79  

[112650 rows x 6 columns]

Podemos analisar a quantidade de itens e o total do valor vendido por cada vendedor.

In [251]:
display(df_items.groupby('ID Vendedor').sum().sort_values('Preço', ascending = False))

<ipython-input-251-ca4e4660f165>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  display(df_items.groupby('ID Vendedor').sum().sort_values('Preço', ascending = False))


Qtd Itens      Preço     Frete
ID Vendedor                                                     
4869f7a5dfa277a7dca6462dcf3b52b2       1193  229472.63  20168.07
53243585a1d6dc2643021fd1853d8905        504  222776.05  13080.63
4a3ca9315b744ce9f8e9374361493884       2269  200472.92  35067.04
fa1c13f2614d7b5c4749cbc52fecda94        594  194042.03  10042.70
7c67e1448b00f6e969d365cea6b010ab       2172  187923.89  51612.55
...                                     ...        ...       ...
34aefe746cd81b7f3b23253ea28bef39          1       8.00     14.52
702835e4b785b67a084280efca355756          1       7.60     10.96
1fa2d3def6adfa70e58c276bb64fe5bb          5       6.90      9.00
77128dec4bec4878c37ab7d6169d6f26          1       6.50      8.72
cf6f6bc4df3999b9c6440f124fb2f687          1       3.50      8.72

[3095 rows x 3 columns]

In [252]:
print(f"A quantidade máxima de produtos vendidos por ordem é: {df_items['Qtd Itens'].max()} produtos.")
print(f"A quantidade mínima de produtos vendidos por ordem é: {df_items['Qtd Itens'].min()} produto.")
print(f"A quantidade média de produtos vendidos por ordem é: {round(df_items['Qtd Itens'].mean(),2)} produto.")

A quantidade máxima de produtos vendidos por ordem é: 21 produtos.
A quantidade mínima de produtos vendidos por ordem é: 1 produto.
A quantidade média de produtos vendidos por ordem é: 1.2 produto.


In [253]:
print(f"O preço máximo dos produtos vendidos por ordem é: R$ {df_items['Preço'].max()}.")
print(f"O preço mínimo dos produtos vendidos por ordem é: R$ {df_items['Preço'].min()}.")
print(f"O preço médio dos produtos vendidos por ordem é: R$ {round(df_items['Preço'].mean(),2)}.")

O preço máximo dos produtos vendidos por ordem é: R$ 6735.0.
O preço mínimo dos produtos vendidos por ordem é: R$ 0.85.
O preço médio dos produtos vendidos por ordem é: R$ 120.65.


In [255]:
print(f"O preço máximo do frete por ordem é: R$ {df_items['Frete'].max()}.")
print(f"O preço mínimo do frete por ordem é: R$ {df_items['Frete'].min()}.")
print(f"O preço médio do frete por ordem é: R$ {round(df_items['Frete'].mean(),2)}.")

O preço máximo do frete por ordem é: R$ 409.68.
O preço mínimo do frete por ordem é: R$ 0.0.
O preço médio do frete por ordem é: R$ 19.99.


Portanto, é possível notar que há ordens que tem frete grátis. Visto que o valor mínimo do frete é R$ 0.00

## Tabela Pagamentos

In [257]:
display(df_payments)
print(f"Shape: {df_payments.shape}")
print(f"Tipo: {type(df_payments)}")
print(f"Tamanho em bytes: {df_payments.to_numpy().nbytes}")

order_id  payment_sequential payment_type  \
0       b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1       a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2       25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3       ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4       42fdf880ba16b47b59251dd489d4441a                   1  credit_card   
...                                  ...                 ...          ...   
103881  0406037ad97740d563a178ecc7a2075c                   1       boleto   
103882  7b905861d7c825891d6347454ea7863f                   1  credit_card   
103883  32609bbb3dd69b3c066a6860554a77bf                   1  credit_card   
103884  b8b61059626efa996a60be9bb9320e10                   1  credit_card   
103885  28bbae6599b09d39ca406b747b6632b1                   1       boleto   

        payment_installments  payment_value  
0                          8          99.33  
1                          1          24.39  
2                          1          65.71  
3                          8         107.78  
4                          2         128.45  
...                      ...            ...  
103881                     1         363.31  
103882                     2          96.80  
103883                     1          47.77  
103884                     5         369.54  
103885                     1         191.58  

[103886 rows x 5 columns]

Shape: (103886, 5)
Tipo: <class 'pandas.core.frame.DataFrame'>
Tamanho em bytes: 4155440


Este DataFrame, possui informações sobre o ID das ordens, o tipo de pagamento, valor do pagamento.

In [258]:
display(df_payments['payment_type'].unique())

array(['credit_card', 'boleto', 'voucher', 'debit_card', 'not_defined'],
      dtype=object)

Com o resultado da saída acima, é possível perceber que os pagamentos utilizados pelos clientes são: Cartão de crédito, boleto, voucher, cartão de débito.

In [267]:
display(df_payments.drop(["order_id", "payment_sequential", "payment_installments",
                          "payment_value"], axis = 1, inplace = False).groupby('payment_type').value_counts().sort_values(ascending = False))

payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
dtype: int64

Portanto, o método de pagamento mais utilizado pelos clientes é o cartão de crédito, seguido pelo boleto, depois voucher e por fim cartão de débito.